inference-engine/tests/functional/shared_test_classes/src/single_layer/binary_convolution.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/single_layer/binary_convolution.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string BinaryConvolutionLayerTest::getTestCaseName(testing::TestParamInfo<binaryConvolutionTestParamsSet> obj) {
+    binConvSpecificParams binConvParams;
+    InferenceEngine::Precision netPrecision;
+    InferenceEngine::Precision inPrc, outPrc;
+    InferenceEngine::Layout inLayout, outLayout;
+    InferenceEngine::SizeVector inputShape;
+    std::string targetDevice;
+
+    std::tie(binConvParams, netPrecision, inPrc, outPrc, inLayout, outLayout, inputShape, targetDevice) = obj.param;
+
+    ngraph::op::PadType padType;
+    InferenceEngine::SizeVector kernel, stride, dilation;
+    std::vector<ptrdiff_t> padBegin, padEnd;
+    size_t convOutChannels;
+    float padValue;
+    std::tie(kernel, stride, padBegin, padEnd, dilation, convOutChannels, padType, padValue) = binConvParams;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShape) << "_";
+    result << "KS=" << CommonTestUtils::vec2str(kernel) << "_";
+    result << "S=" << CommonTestUtils::vec2str(stride) << "_";
+    result << "PB=" << CommonTestUtils::vec2str(padBegin) << "_";
+    result << "PE=" << CommonTestUtils::vec2str(padEnd) << "_";
+    result << "D=" << CommonTestUtils::vec2str(dilation) << "_";
+    result << "O=" << convOutChannels << "_";
+    result << "AP=" << padType << "_";
+    result << "PV=" << padValue << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "inPRC=" << inPrc.name() << "_";
+    result << "outPRC=" << outPrc.name() << "_";
+    result << "inL=" << inLayout << "_";
+    result << "outL=" << outLayout << "_";
+    result << "trgDev=" << targetDevice;
+    return result.str();
+}
+
+InferenceEngine::Blob::Ptr BinaryConvolutionLayerTest::GenerateInput(const InferenceEngine::InputInfo &info) const {
+    InferenceEngine::Blob::Ptr blobPtr;
+    const std::string name = info.name();
+    if (name == "a_data_batch") {
+        blobPtr = FuncTestUtils::createAndFillBlob(info.getTensorDesc(), 1, 0, 1, 7235346);
+    }
```

Comment:
Please make a comment that there is no filters input because with current CPU implementation it is supported only as Constant. Also please make a low priority ticket for CPU plugin team to enable BinaryConvolution to work with Parameter.

Reply:
Done